From 86060701fa7bb4c4c162a4cebbf5a3e521732f39 Mon Sep 17 00:00:00 2001
From: Evy Garden <evysgarden@protonmail.com>
Date: Sun, 15 Oct 2023 18:51:44 +0200
Subject: [PATCH] remove put args sugaring

---
 src/app.rs | 11 ++---------
 1 file changed, 2 insertions(+), 9 deletions(-)

diff --git a/src/app.rs b/src/app.rs
index 63fc101..8980c09 100644
--- a/src/app.rs
+++ b/src/app.rs
@@ -2,7 +2,6 @@ mod command;
 
 use anyhow::Result;
 use clap::Parser;
-use command::put;
 use command::Command;
 
 #[derive(Debug, Parser)]
@@ -16,13 +15,10 @@ use command::Command;
 pub struct Args {
     /// The command to run.
     #[clap(subcommand)]
-    pub command: Option<Command>,
+    pub command: Command,
 
     #[clap(flatten)]
     config_args: ConfigArgs,
-
-    #[clap(flatten)]
-    put_args: put::PutArgs,
 }
 
 #[derive(Debug, Parser)]
@@ -97,10 +93,7 @@ pub enum TimeDisplayMode {
 
 impl Args {
     pub fn run(self) -> Result<()> {
-        match self.command {
-            None => self.put_args.run(&self.config_args)?,
-            Some(command) => command.run(&self.config_args)?,
-        }
+        self.command.run(&self.config_args)?;
         Ok(())
     }
 }
